### PR TITLE
Signal handler improved

### LINKED ISSUE #16: Improve process termination when changes are detected
Many stacks have ON_CHANGE commands that end the process by themselves and there is no need to kill them.

Currently the controller tries to kill them in this case, and prints out warning messages which say the kill failed and there is no such process

Outcome desired:
Either:
1.  suppress the above messages

or 

*2. Use a similar technique that we use to determine server failures and restarts in the on change events where we send a benign signal to determine if the process is alive.
Then we could avoid unnecessary kill actions.

### PR DESCRIPTION
**Pull request by kewegner**
Handle signals gracefully shutting down server and ON_CHANGE processes, and reaping any grand child processes caused by the shutdown.

Eliminate unncessary kill messages when there is no ON_CHANGE process to kill

Fixes #16

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -709,3 +749,38 @@ func main() {
...
+		pid, err := syscall.Wait4(-1, &wstatus, syscall.WNOHANG, nil)
+		ControllerDebug.log("Reaper pid/err is: ", pid, err)
+		if pid == 0 && countLimit < maxLimit && err == nil {
+			ControllerDebug.log("Reaper sleeping 1 second: ", pid)
+			time.Sleep(1 * time.Second)
+			countLimit++
+		}
+
+		if syscall.EINTR == err {
+			ControllerDebug.log("Signal Interupt: ", err)
```
Comment (chilanti) [inline main.go]: There's a typo here. "Interrupt".

File: main.go [hunk 2]
```diff
@@ -695,6 +709,32 @@ func main() {
...
 		go runCommands(startCommand, server, false, false)
 	}
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM, syscall.SIGINT)
+	go func() {
+		<-c
+		cmps.mu.Lock()
+		defer cmps.mu.Unlock()
+		ControllerInfo.log("Inside signal handler for controller")
```
Comment (chilanti) [inline main.go]: Should this be ControllerDebug?

Comment (kylegc) [inline main.go]: I think all of these Info messages in the signal handler should be Debug level.

Maybe add just one Info message like "Interrupt signal received, shutting down processes."

Comment (kewegner) [reply main.go]: changes made

Comment (kewegner) [reply main.go]: changed

**Pull request merged**